.github/workflows/test_security.yml

```diff
@@ -0,0 +1,96 @@
+name: Security test workflow for k-NN
+on:
```

Comment:
we should also make this as periodic github action.

Reply:
ack